lib/fencing/st_client.c

```diff
@@ -1336,32 +1353,29 @@ xml_to_event(xmlNode * msg)
 {
     stonith_event_t *event = calloc(1, sizeof(stonith_event_t));
     const char *ntype = crm_element_value(msg, F_SUBTYPE);
-    char *data_addr = crm_strdup_printf("//%s", ntype);
-    xmlNode *data = get_xpath_object(data_addr, msg, LOG_DEBUG);
 
     crm_log_xml_trace(msg, "stonith_notify");
 
     crm_element_value_int(msg, F_STONITH_RC, &(event->result));
 
     if (pcmk__str_eq(ntype, T_STONITH_NOTIFY_FENCE, pcmk__str_casei)) {
-        event->operation = crm_element_value_copy(msg, F_STONITH_OPERATION);
+        xmlNode *data = get_event_data_xml(msg, ntype);
```

Comment:
wouldn't we just be fine with
xmlNode *data = get_xpath_object("//" T_STONITH_NOTIFY_FENCE, msg, LOG_DEBUG);
unless of course it is reused with a later change.

Reply:
I haven't reused it yet, but I thought functionizing helped de-clutter by hiding the string alloc/free.

We can't hardcode T_STONITH_NOTIFY_FENCE because other types are used, they just have much less info.

Reply:
at least for that code-block it is assured that ntype == T_STONITH_NOTIFY_FENCE.
as said if we intend to use it anywhere else having it generic and functionalized is fine.

Reply:
Now I remember, I thought the other types have too little info currently, and we might want to address that at some point, and this could come in handy then